### PR TITLE
Lab 3 - Letter Frequency

### LINKED ISSUE #346: Lab 3 - Letter Count
- [x] Build map of letters and their frequencies - `letters(string) map[rune]int`
- [x] Build a sorted slice of letter-frequency strings - `sortLetters(m map[rune]int) []string`
- [x] Print letter frequency report for "aba" from Main

### PR DESCRIPTION
**Pull request by marksomething**
Fixes #346

Review of colleague's PR #333

#### Changes proposed in this PR:
* letters(string) map[rune]int
  returns a map of the frequency of runes
* sortLetters(map[rune]int) [] string
  returns a slice of rune:frequency pairs formatted as string
  and ordered by descending frequency
* main()
  prints an ordered list of rune:freuency pairs

### CODE REVIEW

**Review by pfannerj — COMMENTED** [review]
Your solution for the letters function is much simpler than mine... I created the array of runes separately before looping through to count.  This is really great!  Only suggestion I have would be to give 'v' a meaningful name.

The test cases also look good and testing for emojis is wise too.  It got me thinking though... I thought that the sortLetters function needed to return the list in rune order and not in the order of most to least??... I checked some other merged solutions which would return a:1 b:2 for bba... need to check this again on slack I guess... 
If I'm correct then think it would allow your sortLetters solution to be simplified too...

File: 03_letters/marksomething/main.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	// Convert to slice
+	var ltrCounts = []ltrCount{}
+	for r, count := range m {
+		ltrCounts = append(ltrCounts, ltrCount{r, count})
+	}
+
+	lessFn := func(i int, j int) bool {
+		// Secondary sort by rune to make result deterministic
```
Comment (camscale) [inline 03_letters/marksomething/main.go:34]: Good attention to detail.

**Review by camscale — DISMISSED** [review]
This is excellent Go code. It is clean and comprehensive. Thank you.

Sorry for the review/approval delay. Please ping me on slack (@camh) if you need to rebase and need re-approval. I'll do that immediately.

**Comment by camh** [comment]
Hey, I think you’re accidentally tagging me in some PRs, @camh-anz

**Comment by camscale** [comment]
> Hey, I think you’re accidentally tagging me in some PRs, @camh-anz

Sorry @camh - was using my slack username - didn't think of the consequences. Do you know I'm jealous you have my username on github? I'll take more care in future. (and sorry @marksomething for carrying on an unrelated conversation on your PR).

**Review by camscale — APPROVED** [review]
LGTM for rebase

**Pull request merged**